src/plugins/ImportPlugin.js

```diff
@@ -0,0 +1,107 @@
+import { $injector } from '../injection';
+import { addLayer } from '../store/layers/layers.action';
+import { emitNotification, LevelTypes } from '../store/notifications/notifications.action';
+import { VectorSourceType } from '../services/domain/geoResources';
+import { observe } from '../utils/storeUtils';
+import { provide as provider } from './i18n/importPlugin.provider';
+import { BaPlugin } from './BaPlugin';
+import { MediaType } from '../services/HttpService';
+import { SourceTypeName } from '../services/SourceTypeService';
+
+/**
+ * @class
+ * @author thiloSchlemmer
+ */
+export class ImportPlugin extends BaPlugin {
+	constructor() {
+		super();
+		const { ImportVectorDataService: importVectorDataService, SourceTypeService: sourceTypeService, TranslationService: translationService } = $injector.inject('ImportVectorDataService', 'SourceTypeService', 'TranslationService');
+		this._importVectorDataService = importVectorDataService;
+		this._sourceTypeService = sourceTypeService;
+		this._translationService = translationService;
+		translationService.register('importPluginProvider', provider);
+	}
+
+	/**
+	 * @override
+	 * @param {Store} store
+	 */
+	async register(store) {
+
+		const onChange = async (latestImport) => {
+			const { payload: { url, data, mimeType } } = latestImport;
+			try {
+				const vectorGeoResource = url ? await this._importByUrl(url) : this._importByData(data, mimeType);
+				if (vectorGeoResource) {
+					const { id, label } = vectorGeoResource;
+					addLayer(id, { label: label });
+				}
+			}
+			catch (error) {
+				emitNotification(error, LevelTypes.ERROR);
+			}
+		};
+
+		observe(store, state => state.import.latest, onChange);
+	}
+
+	/**
+	 * Imports the data as remote {@link GeoResource}
+	 * @param {string} url the url to the data
+	 * @returns {Promise<GeoResource>} the imported GeoResource
+	 */
+	async _importByUrl(url) {
+		const sourceType = await this._sourceTypeService.forURL(url);
+		const createGeoResource = (url, sourceType) => {
+			if (sourceType) {
+				switch (sourceType.name) {
+					case SourceTypeName.KML:
+					case SourceTypeName.GPX:
+					case SourceTypeName.GEOJSON:
+						return this._importVectorDataService.importVectorDataFromUrl(url);
```

Comment:
Here we have to set the `VectorSourceType` within the options => map sourceType to vectorSourceType